core/datatrak.c

```diff
@@ -26,13 +27,9 @@ static unsigned long four_bytes_to_long(unsigned char x, unsigned char y, unsign
 	return ((long)x << 24) + ((long)y << 16) + ((long)z << 8) + (long)t;
 }
 
-static unsigned char *byte_to_bits(unsigned char byte_value)
+static bool bit_set(unsigned char byte, int bit)
```

Comment:
Does the compiler figure this is a good candidate for inlining? What about making the arguments static (once you start worrying about these kinds of things...)?

Reply:
> Does the compiler figure this is a good candidate for inlining?

Compiles down to a single instruction for me (gcc with -O2):

```
        .loc 1 32 2 discriminator 14 view .LVU462
        .loc 1 361 5 is_stmt 0 discriminator 14 view .LVU463
        testb   $4, %r15b
        jne     .L238
```

Reply:
hehe - I'd consider that fairly decent optimization :-)